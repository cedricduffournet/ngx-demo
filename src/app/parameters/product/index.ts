export * from './parameter-product.component';
