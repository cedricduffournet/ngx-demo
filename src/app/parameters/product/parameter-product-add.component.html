<app-product-add-view></app-product-add-view>
