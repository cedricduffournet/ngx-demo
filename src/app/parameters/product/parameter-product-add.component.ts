import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-product-add',
  templateUrl: './parameter-product-add.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterProductAddComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
