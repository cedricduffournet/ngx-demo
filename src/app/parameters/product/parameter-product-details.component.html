<app-product-details-view></app-product-details-view>
