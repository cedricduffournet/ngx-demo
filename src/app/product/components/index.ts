export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
