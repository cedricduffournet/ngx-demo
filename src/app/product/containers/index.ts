export * from './product-add-view/product-add-view.component';
export * from './product-update-modal/product-update-modal.component';
export * from './product-delete-modal/product-delete-modal.component';
export * from './product-list-view/product-list-view.component';
export * from './product-selected/product-selected.component';
export * from './product-details-view/product-details-view.component';
