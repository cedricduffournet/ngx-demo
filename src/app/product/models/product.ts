import { schema } from 'normalizr';

export interface Product {
  id: number;
  name: string;
}

export const productSchema = new schema.Entity('products');
