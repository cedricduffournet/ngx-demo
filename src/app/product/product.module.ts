import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { ProductEffects } from '@app/product/state/effects';
import { ProductFacade } from '@app/product/state/product.facade';
import {
  ProductAddComponent,
  ProductUpdateComponent,
  ProductDeleteComponent,
  ProductFormComponent,
  ProductItemComponent,
  ProductItemsComponent
} from '@app/product/components';
import {
  ProductAddModalComponent,
  ProductUpdateModalComponent,
  ProductDeleteModalComponent,
  ProductListViewComponent
} from '@app/product/containers';
import { reducers } from '@app/product/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([ProductEffects]),
    StoreModule.forFeature('products', reducers)
  ],
  declarations: [
    ProductListViewComponent,
    ProductAddComponent,
    ProductUpdateComponent,
    ProductDeleteComponent,
    ProductFormComponent,
    ProductItemComponent,
    ProductItemsComponent,
    ProductAddModalComponent,
    ProductUpdateModalComponent,
    ProductDeleteModalComponent
  ],
  entryComponents: [
    ProductUpdateModalComponent,
    ProductAddModalComponent,
    ProductDeleteModalComponent
  ],
  exports: [ProductListViewComponent],
  providers: [ProductFacade]
})
export class ProductModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
