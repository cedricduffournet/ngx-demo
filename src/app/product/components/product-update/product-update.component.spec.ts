import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import {
  ProductUpdateComponent,
  ProductFormComponent
} from '@app/product/components';
import { Product } from '@app/product/models/product';

describe('ProductUpdate', () => {
  let fixture: ComponentFixture<ProductUpdateComponent>;
  let component: ProductUpdateComponent;
  const product: Product = {
    id: 1,
    name: 'testname1'
  } as Product;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule
      ],
      declarations: [ProductUpdateComponent, ProductFormComponent]
    });

    fixture = TestBed.createComponent(ProductUpdateComponent);
    component = fixture.componentInstance;
    component.product = product;
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit save event', () => {
    spyOn(component.update, 'emit');
    component.onSave({ name: 'nameupdated' } as Product);
    expect(component.update.emit).toHaveBeenCalledWith({
      id: product.id,
      product: { name: 'nameupdated' }
    });
  });

  it('should indicate product is updating', () => {
    component.updating = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit cancel event', () => {
    spyOn(component.cancel, 'emit');
    component.onCancel();
    expect(component.cancel.emit).toHaveBeenCalledWith('cancel');
  });
});
