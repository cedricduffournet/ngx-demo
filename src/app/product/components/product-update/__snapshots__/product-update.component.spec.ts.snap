// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductUpdate should be created 1`] = `
<app-product-update
  cancel={[Function EventEmitter]}
  product={[Function Object]}
  update={[Function EventEmitter]}
  updating="false"
>
  <app-product-form
    ng-reflect-processing="false"
    ng-reflect-product="[object Object]"
  >
    <form
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="false"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              <button
                class="btn btn-cancel btn-lg btn-light"
                ng-reflect-klass="btn btn-cancel btn-lg"
                ng-reflect-ng-class="btn-light"
                type="button"
              >
                 CANCEL 
              </button>
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                
                
                <span>
                  SAVE
                </span>
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-product-form>
</app-product-update>
`;

exports[`ProductUpdate should indicate product is updating 1`] = `
<app-product-update
  cancel={[Function EventEmitter]}
  product={[Function Object]}
  update={[Function EventEmitter]}
  updating={[Function Boolean]}
>
  <app-product-form
    ng-reflect-processing="true"
    ng-reflect-product="[object Object]"
  >
    <form
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <section
        class="bg-white"
      >
        <div
          class="p-3"
        >
          <app-text-input
            class="ng-untouched ng-pristine ng-valid"
            formcontrolname="name"
            ng-reflect-label="NAME"
            ng-reflect-name="name"
            ng-reflect-submitted="false"
          >
            <div
              class="form-group "
            >
              <label
                class="control-label"
              >
                NAME
              </label>
              <input
                class="form-control"
                placeholder=""
                type="text"
              />
              
              
              
              
              
              
            </div>
          </app-text-input>
        </div>
      </section>
      <app-wrapper-validation-action>
        <div
          class="d-flex justify-content-end align-items-center p-3"
        >
          <app-validation-button
            ng-reflect-confirm-disabled="true"
            ng-reflect-label-cancel="CANCEL"
            ng-reflect-label-confirm="SAVE"
          >
            <div
              class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
              ng-reflect-klass="button-validation-action text-"
              ng-reflect-ng-class=""
            >
              
              
              <button
                class="btn btn-confirm btn-lg btn-primary"
                disabled=""
                ng-reflect-klass="btn btn-confirm btn-lg"
                ng-reflect-ng-class="btn-primary"
                type="submit"
              >
                
                <span
                  class="spinner-border spinner-border-sm align-baseline"
                  style="font-size: 1rem;"
                />
                
                <span>
                  Â LOADING
                </span>
                
              </button>
            </div>
          </app-validation-button>
        </div>
      </app-wrapper-validation-action>
    </form>
  </app-product-form>
</app-product-update>
`;
